python/foglamp/services/core/api/plugins/update.py

```diff
@@ -39,48 +40,76 @@
         curl -sX PUT http://localhost:8081/foglamp/plugins/south/sinusoid/update
         curl -sX PUT http://localhost:8081/foglamp/plugins/north/http_north/update
         curl -sX PUT http://localhost:8081/foglamp/plugins/filter/metadata/update
+        curl -sX PUT http://localhost:8081/foglamp/plugins/notificationDelivery/asset/update
+        curl -sX PUT http://localhost:8081/foglamp/plugins/notificationRule/OutOfBound/update
     """
     _type = request.match_info.get('type', None)
     name = request.match_info.get('name', None)
     try:
-        # TODO: FOGL-3064
-        _type = _type.lower()
-        if _type not in ['north', 'south', 'filter']:
-            raise ValueError("Invalid plugin type. Must be 'north', 'south', 'filter'")
+        _type = _type.lower() if not str(_type).startswith('notification') else _type
+        if _type not in ['north', 'south', 'filter', 'notificationDelivery', 'notificationRule']:
+            raise ValueError("Invalid plugin type. Must be 'north', 'south', 'filter', 'notificationDelivery' "
+                             "or 'notificationRule'")
 
         # Check requested plugin name is installed or not
         installed_plugins = PluginDiscovery.get_plugins_installed(_type, False)
         installed_plugin_name = [p_name["name"] for p_name in installed_plugins]
         if name not in installed_plugin_name:
             raise KeyError("{} plugin is not yet installed. So update is not possible.".format(name))
 
-        # Tracked plugins from asset tracker
-        tracked_plugins = await _get_plugin_and_sch_name_from_asset_tracker(_type)
         sch_list = []
-        filters_used_by = []
-        if _type == 'filter':
-            # In case of filter, for asset_tracker table we are inserting filter category_name in plugin column
-            # instead of filter plugin name by Design
-            # Hence below query is required to get actual plugin name from filters table
+        notify_list = []
+        if _type in ['notificationDelivery', 'notificationRule']:
+            # Check Notification service is enabled or not
+            payload = PayloadBuilder().SELECT("id", "enabled", "schedule_name").WHERE(['process_name', '=',
```

Comment:
can not there be multiple schedules of notification_c, (may be disabled)? This is important to fix as we are checking sch_info[0].

Reply:
Nope, we have added a restriction of notification service with one instance only when hit  add_service endpoint

Reply:
Okay